### DISCUSSION TITLE
System plotting configuration

### DISCUSSION THREAD

**Discussion by DgrMouse** [rdudhagra/Sand-Table#21] [Q&A] [answered]
Is it easy enough to convert the system from a Cartesian plotting system to a Polar plotting system?

**Comment by rdudhagra** [comment, marked as answer]
It depends. 

Hardware-wise, you'll probably need a complete redesign of the gantry and maybe the table if you want it to be circular. However, there are lots of designs on the internet for round tables that you could probably grab/get inspiration from.

If you already have a polar plotting system built, then it's easier to convert; however, you will have to modify some of the code. Marlin (the 3D Printer firmware that the sane table uses, doesn't support polar coordinates natively as I know (unless you find a fork/alternative that does...that would be the easiest solution); so, we will have to modify the web server. Here's a rough overview but let me know if you have additional questions:
1. In `process_theta_rho.js`, most of the code here will need to be rewritten; however, the replacement will be fairly short and simple. Instead of converting each polar coordinate to cartesian, we'll need to keep them as polar coordinates, but still write them in terms of `x` and `y`. So, each line of the GCODE file will need to look like: `G0 X[the "r" value from the .thr file] Y[the "theta" value from the .thr file]`. Take a look at `printRectangularCommand()` to actually generate the strings/etc...you'll only need to modify parts of the code that call that function, such as `printArcCommand()`. 
2. Change the Marlin firmware to set the printer to `Cartesian` mode rather than CoreXY. You may also need to retune sensorless homing parameters if you're using the TMC2209's like I did.
3. Hook up the motor for the radius of your polar plotting system to the `X` channel, and the rotation motor to the `Y` channel. (You can flip these/etc as long as you're consistent of course)

Let me know if that helps...it's been a while since I've looked at the firmware for this project but let me know if you have questions about what you'd need to modify and I can try to dig deeper.
